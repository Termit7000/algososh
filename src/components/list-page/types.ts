export type TStore<T=string> = {
    data: (T | '')[],
    direction: 'ADDING' | 'REMOVING' | '',
    inProgress: number[],
    modified: number[],
    headIndex: number,
    tailIndex: number,
    insertion: {
        value: T | '',
        index: number
    }    
}