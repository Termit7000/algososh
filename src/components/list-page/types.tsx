export type TStore = {
    data: string[],
    direction: 'ADDING' | 'REMOVING' | '',
    inProgress: number[],
    modified: number[],
    headIndex: number,
    tailIndex: number,
    insertion: {
        value: string,
        index: number
    }    
}