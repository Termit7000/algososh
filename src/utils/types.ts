export const ASC = 'ASC';
export const DESC = 'DESC';

export type TSortKind = typeof ASC | typeof DESC;