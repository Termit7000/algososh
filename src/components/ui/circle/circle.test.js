import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom'

import { Circle } from './circle';
import { ElementStates } from '../../../types/element-states';

describe("Circle", ()=>{

    it("Без буквы", ()=>{
        const circle = <Circle/>;        
        render(circle);
        expect(screen.getByTestId('circle')).toMatchSnapshot();
    })

    it("С буквой", ()=>{
        const circle = <Circle letter='BFG'/>;        
        render(circle);
        expect(screen.getByTestId('circle')).toMatchSnapshot();
    })

    it("С head", ()=>{
        const circle = <Circle head={'hdd'}/>;        
        render(circle);        
        expect(screen.getByTestId('circle')).toMatchSnapshot();
    })

    it("С react-элемент в head", ()=>{
        const head = <p data-testid='head'>head</p>;
        const circle = <Circle head={head}/>;        
        render(circle);        
        expect(screen.getByTestId('circle')).toMatchSnapshot();
    })

    it("С tail", ()=>{        
        const circle = <Circle tail={'TLL'}/>;        
        render(circle);        
        expect(screen.getByTestId('circle')).toMatchSnapshot();
    })

    it("С react-элемент в tail", ()=>{        
        const tail = <p data-testid='tail'>tail</p>;
        const circle = <Circle tail={tail}/>;        
        render(circle);        
        expect(screen.getByTestId('circle')).toMatchSnapshot();
    })

    it("С index", ()=>{        
        const circle = <Circle index={10200}/>;        
        render(circle);        
        expect(screen.getByTestId('circle')).toMatchSnapshot();
    })

    it("С пропом isSmall===true", ()=>{        
        const circle = <Circle isSmall={true}/>;        
        render(circle);        
        expect(screen.getByTestId('circle')).toMatchSnapshot();
    })

    it("В состоянии default", ()=>{        
        const circle = <Circle state={ElementStates.Default}/>;        
        render(circle);        
        expect(screen.getByTestId('circle')).toMatchSnapshot();
    })

    it("В состоянии Changing", ()=>{        
        const circle = <Circle state={ElementStates.Changing}/>;        
        render(circle);        
        expect(screen.getByTestId('circle')).toMatchSnapshot();
    })

    it("В состоянии Modified", ()=>{        
        const circle = <Circle state={ElementStates.Modified}/>;        
        render(circle);        
        expect(screen.getByTestId('circle')).toMatchSnapshot();
    })

})