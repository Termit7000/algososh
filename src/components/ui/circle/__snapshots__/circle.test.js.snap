// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle Без буквы 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="circle_value"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle В состоянии Changing 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   changing"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="circle_value"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle В состоянии Modified 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   modified"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="circle_value"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle В состоянии default 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="circle_value"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle С head 1`] = `
<div
  class="text text_type_input text_color_input mb-4 absolute head string"
>
  hdd
</div>
`;

exports[`Circle С index 1`] = `
<p
  class="text text_type_input text_color_input mt-4 absolute index"
>
  10200
</p>
`;

exports[`Circle С react-элемент в head 1`] = `
<p
  data-testid="head"
>
  head
</p>
`;

exports[`Circle С react-элемент в tail 1`] = `
<p
  data-testid="tail"
>
  tail
</p>
`;

exports[`Circle С tail 1`] = `
<div
  class="text text_type_input text_color_input mt-4 absolute tail30 string"
>
  TLL
</div>
`;

exports[`Circle С буквой 1`] = `
<p
  class="text text_type_circle text_color_input letter"
  data-testid="circle_value"
>
  BFG
</p>
`;

exports[`Circle С пропом isSmall===true 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle  small default"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="circle_value"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
